src/platform/macos.rs

```diff
@@ -0,0 +1,207 @@
+//! The use of NSVisualEffectView comes from https://github.com/joboet/winit/tree/macos_blurred_background
+//! just rewrite it to make it more like cocoa::appkit style .
+
+#![cfg(target_os = "macos")]
+
+use cocoa::{
+    appkit::{
+        NSAppKitVersionNumber, NSAppKitVersionNumber10_10, NSAppKitVersionNumber10_11,
+        NSAutoresizingMaskOptions, NSView, NSViewHeightSizable, NSViewWidthSizable, NSWindow,
+        NSWindowOrderingMode,
+    },
+    base::{id, nil, BOOL},
+    foundation::{NSAutoreleasePool, NSPoint, NSRect, NSSize},
+};
+use objc::{class, msg_send, sel, sel_impl};
+
+#[allow(deprecated)]
+pub fn apply_blur(window: id) {
```

Comment:
```suggestion
pub fn apply_vibrancy(window: id, appearance: MacVibrancy) {
```
where `MacVibrancy` is just an export of `NSVisualEffectMaterial` enum under a new name.